src/main/java/uk/gov/companieshouse/confirmationstatementapi/client/OracleQueryClient.java

```diff
@@ -167,21 +169,20 @@ public boolean isConfirmationStatementPaid(String companyNumber, String dueDate)
         return confirmationStatementPaymentJson.isPaid();
     }
 
-    public String getRegisteredEmailAddress(String companyNumber) throws ServiceException {
+    public String getRegisteredEmailAddress(String companyNumber) throws ServiceException, RegisteredEmailNotFoundException {
         var url = String.format("%s/company/%s/registered-email-address", oracleQueryApiUrl, companyNumber);
         ApiLogger.info(String.format(CALLING_ORACLE_QUERY_API_URL_GET, url));
 
-        ResponseEntity<RegisteredEmailAddressJson> response = restTemplate.getForEntity(url, RegisteredEmailAddressJson.class);
-
-        if (response.getStatusCode() == HttpStatus.OK) {
-            var body = response.getBody();
-            if (body != null) {
-                return body.getRegisteredEmailAddress();
-            } else {
-                throw new ServiceException(ORACLE_QUERY_API_NO_DATA);
-            }
-        } else {
-            throw new ServiceException(String.format(ORACLE_QUERY_API_STATUS_MESSAGE, response.getStatusCode(), companyNumber));
+        ResponseEntity<RegisteredEmailAddressJson> response;
+        try {
+            response = restTemplate.getForEntity(url, RegisteredEmailAddressJson.class);
+        } catch (HttpClientErrorException e) {
```

Comment:
I'd expect to see you checking the status code of the exception here & handling 404 differently from other errror codes.

Reply:
The original code construct could be extended to handle the 404 (as specified in the Swagger for the oracle-query-api).

Note the Swagger for this service also need to be updated to be consistent with your changes where now this service is also going to generate 404 for a blank/null email :-(

```
        ResponseEntity<RegisteredEmailAddressJson> response = restTemplate.getForEntity(url, RegisteredEmailAddressJson.class);

        if (response.getStatusCode() == HttpStatus.OK) {
            var body = response.getBody();
            if (body != null) {
                return body.getRegisteredEmailAddress();
            } else {
                throw new ServiceException(ORACLE_QUERY_API_NO_DATA);
            }
        } else if (response.getStatusCode() == HttpStatus.NOT_FOUND) {
            / *
             * Sadly this is the HTTP status code currently defined in Swagger for oracle-query-api service when the email is blank/null.
             * The misuse is going to be mirrored in this service, i.e. for now keep the APIs in CH consistent rather than fixing.
             * /
            throw new RegisteredEmailNotFoundException(REGISTERED_EMAIL_ADDRESS_NOT_FOUND);
        } else {
            throw new ServiceException(String.format(ORACLE_QUERY_API_STATUS_MESSAGE, response.getStatusCode(), companyNumber));
```

Reply:
getForEntity throws an HttpClientErrorException